File: LeetCode/palindrome_number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x < 0)
            return false;
        
        int rev = reverse(x);
        if(rev == x)
            return true;
        return false;
    }
    
    int reverse(int x) {
        long long int aux = 0;
        while(x != 0) {
            aux = aux*10 + x%10;
            x /= 10;
        }
        return aux;
    }
    
};

File: LeetCode/palindrome_number.go
func isPalindrome(x int) bool {
    if x < 0 {
        return false
    }

    reversed, aux := 0, x
    for x != 0 {
        reversed = (reversed * 10) + (x % 10)
        x = x / 10
    }

    return reversed == aux
}
